packages/shared/src/cssVars.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Normalize CSS var value created by `v-bind` in `<style>` block
+ * See https://github.com/vuejs/core/pull/12461#issuecomment-2495804664
+ */
+export function normalizeCssVarValue(value: unknown): string {
+  if (value == null) {
+    return 'initial'
+  }
+
+  if (typeof value === 'string') {
+    return value === '' ? ' ' : value
+  }
+
+  if (typeof value !== 'number' || !Number.isFinite(value)) {
```

Comment:
Note for reviewers: Apart from the proposed solution at https://github.com/vuejs/core/pull/12461#issuecomment-2495804664, I believe it is reasonable to allow finite numbers to be bound to CSS properties. However, for other data types, I couldn’t determine any clear expected behavior and have added a development-time warning for such cases.